addons/mail/models/mail_channel.py

```diff
@@ -149,25 +267,45 @@ def unlink(self):
 
     @api.multi
     def write(self, vals):
+        # First checks if user tries to modify moderation fields and has not the right to do it.
+        # We don't check if the fields touched are actually modified.
+        moderation_keys_are_touched = any(key for key in moderation_keys if vals.get(key))
+        if moderation_keys_are_touched and not self.env.user.has_group('base.group_system'):
```

Comment:
Just to avoid perform checks in the right order, I would 1/ check for moderators and 2/ check for group_system.

Reply:
done